native/metro.config.js

```diff
@@ -6,55 +6,45 @@
  *
  * @format
  */
+const { getDefaultConfig, mergeConfig } = require('@react-native/metro-config')
 const path = require('path')
-const Resolver = require('metro-resolver')
 
-const defaultSourceExts = require('metro-config/src/defaults/defaults').sourceExts
-const defaultAssetExts = require('metro-config/src/defaults/defaults').assetExts
+/** @type {import('metro-config').MetroConfig} */
+const defaultConfig = getDefaultConfig(__dirname)
+const {
+  resolver: { assetExts, sourceExts },
+} = defaultConfig
 
-module.exports =
-  // https://github.com/facebook/react-native/issues/21310#issuecomment-544071895
-  // https://github.com/facebook/metro/issues/1#issuecomment-453450709
-  // Per default metro is looking for the babel dependencies in the node_modules of the corresponding package.
-  // This leads to problems in the build-configs module as the babel dependencies are only in the native .
-  {
-    resolver: {
-      assetExts: defaultAssetExts.filter(ext => ext !== 'svg'),
-      sourceExts: [...defaultSourceExts, 'svg'],
-      extraNodeModules: new Proxy(
-        {},
-        {
-          get: (target, name) => {
-            if (name === 'build-config-name') {
-              const buildConfigName = process.env.BUILD_CONFIG_NAME || 'integreat-test-cms'
-              // Proxy the (non-existing) module 'build-config-name' to the name of the right build config.
-              // Passing environment variables is not possible without either babel or post processing otherwise.
-              return path.resolve(__dirname, '../build-configs', buildConfigName, 'build-config-name/index.ts')
-            }
-
-            return path.resolve(__dirname, `node_modules/${name}`)
-          },
-        },
+/** @type {import('metro-config').MetroConfig} */
+const config = {
+  resolver: {
+    assetExts: assetExts.filter(ext => ext !== 'svg'),
+    sourceExts: [...sourceExts, 'svg'],
+    extraNodeModules: {
+      'build-config-name': path.resolve(
+        __dirname,
+        '../build-configs',
+        process.env.BUILD_CONFIG_NAME || 'integreat-test-cms',
+        'build-config-name/index.ts',
       ),
-      // Make sure we use the local copy of react and react-native to avoid multiple copies in the bundle
-      // https://github.com/facebook/react/issues/13991#issuecomment-830308729
-      resolveRequest: (context, moduleName, platform) => {
-        const module =
-          moduleName === 'react' || moduleName === 'react-native'
-            ? path.join(__dirname, 'node_modules', moduleName)
-            : moduleName
-        return Resolver.resolve(context, module, platform)
-      },
     },
-    watchFolders: [path.resolve(__dirname, '../')],
-    transformer: {
-      allowOptionalDependencies: true,
-      babelTransformerPath: require.resolve('react-native-svg-transformer'),
-      getTransformOptions: async () => ({
-        transform: {
-          experimentalImportSupport: false,
-          inlineRequires: true,
-        },
```

Comment:
This is set by react native already:
https://github.com/facebook/react-native/blob/main/packages/metro-config/index.js#L70-L82